### PR TITLE
Remove fallback to interpreter for JoinCompiler

### PR DESCRIPTION
**Pull request by dain**

### CODE REVIEW

**Review by martint — APPROVED** [review]

File: presto-main/src/main/java/io/prestosql/operator/PagesIndex.java [hunk 1]
```diff
@@ -493,20 +493,15 @@ public LookupSourceSupplier createLookupSourceSupplier(
...
+                    session,
+                    valueAddresses,
+                    channels,
+                    hashChannel,
+                    filterFunctionFactory,
+                    sortChannel,
+                    searchFunctionFactories);
         }
 
         // if compilation fails
```
Comment (findepi) [inline presto-main/src/main/java/io/prestosql/operator/PagesIndex.java:507]: update this too

File: presto-main/src/main/java/io/prestosql/operator/PagesIndex.java [hunk 2]
```diff
@@ -493,20 +493,15 @@ public LookupSourceSupplier createLookupSourceSupplier(
...
-                        session,
-                        valueAddresses,
-                        channels,
-                        hashChannel,
-                        filterFunctionFactory,
-                        sortChannel,
-                        searchFunctionFactories);
-            }
-            catch (Exception e) {
-                log.error(e, "Lookup source compile failed for types=%s error=%s", types, e);
```
Comment (findepi) [inline presto-main/src/main/java/io/prestosql/operator/PagesIndex.java:508]: same in `io.prestosql.operator.PagesIndex#createPagesHashStrategy(java.util.List<java.lang.Integer>, java.util.OptionalInt, java.util.Optional<java.util.List<java.lang.Integer>>)` 
https://github.com/prestosql/presto/blob/e7199dba8e2551cd6d70cae891c8e2e14200a678/presto-main/src/main/java/io/prestosql/operator/PagesIndex.java#L436-L442
?

**Comment by jerryleooo** [comment]
Hi @dain @findepi may I know the purpose of this PR? Thanks

File: presto-main/src/main/java/io/prestosql/operator/PagesIndex.java (hunk 1 above)
Comment (findepi) [reply presto-main/src/main/java/io/prestosql/operator/PagesIndex.java:507]: addressed in https://github.com/trinodb/trino/pull/10033. thanks @sopel39 for taking care of this :)

**Pull request merged**